### PR TITLE
Http2 adatper header support dynamic

### PR DESCRIPTION
**Pull request by Passer-by**
### New Pull Request Checklist

- [x] I have read the [Documentation](https://pub.dev/documentation/dio/latest/)
- [x] I have searched for a similar pull request in the [project](https://github.com/cfug/dio/pulls) and found none
- [x] I have updated this branch with the latest `main` branch to avoid conflicts (via merge from master or rebase)
- [x] I have added the required tests to prove the fix/feature I'm adding
- [ ] I have updated the documentation (if necessary)
- [x] I have run the tests without failures
- [x] I have updated the `CHANGELOG.md` in the corresponding package

### CODE REVIEW

**Comment by Passer-by** [comment]
Dio's header is of type Map<String, dynamic>, so the Http2Adapter cannot directly convert the value to a string.
But there is a question about this PR: Why does Dio perform a default trim operation on the value? Unlike the native Dart HttpClient, which does not trim when setting headers. Should I adhere to Dio's standard or the Dart HttpClient standard?

**Comment by AlexV525** [comment]
> Should I adhere to Dio's standard or the Dart HttpClient standard?

Thanks for details. I think we should stick to RFCs instead of Dio's. We'll manage to get rid of the Dio's implementation too.

File: plugins/http2_adapter/lib/src/http2_adapter.dart [hunk 1]
```diff
@@ -61,14 +61,18 @@ class Http2Adapter implements HttpClientAdapter {
...
-          .map(
-            (key) => Header.ascii(
-              key.toLowerCase(),
-              options.headers[key] as String? ?? '',
-            ),
-          )
-          .toList(),
+      options.headers.keys.map(
+        (key) {
+          final String v;
```
Comment (AlexV525) [inline plugins/http2_adapter/lib/src/http2_adapter.dart:66]: Use `entries` instead.

File: plugins/http2_adapter/test/http2_test.dart [hunk 2]
```diff
@@ -133,4 +133,28 @@ void main() {
...
+          'NumKey': 2,
+          'BooleanKey': false,
+        },
+      ),
+    );
+    expect(res.data.toString(), contains('TEST'));
+    expect(res.data.toString(), contains('Listkey: 1, 2'));
+    expect(res.data.toString(), contains('Stringkey: 1'));
+    expect(res.data.toString(), contains('Numkey: 2'));
+    expect(res.data.toString(), contains('Booleankey: false'));
```
Comment (AlexV525) [inline plugins/http2_adapter/test/http2_test.dart:158]: These toString() calls can combined into only one variable.

File: plugins/http2_adapter/CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,7 @@
 - Implement `sendTimeout` and `receiveTimeout` for the adapter.
 - Fix redirect not working when requestStream is null.
 - Ignores `Duration.zero` timeouts.
+- `Http2Adapter` header supports dynamic
```
Comment (AlexV525) [inline plugins/http2_adapter/CHANGELOG.md:13]:
```suggestion
- Add more header value types implicit support.
```

Comment (AlexV525) [inline plugins/http2_adapter/CHANGELOG.md:13]: Also, put this in the *Unreleased* section instead of here.

**Review by AlexV525 — APPROVED** [review]

**Pull request merged**